src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -39,6 +60,16 @@ def synthetic_target_extra_dependencies(self, target):
     return []
 
   @property
+  def forced_codegen_strategy(self):
```

Comment:
Maybe the subclass could have the property return a list of supported codegen strategies and then just error out if the option is specifies otherwise.